### DISCUSSION TITLE
Background tasks never being called after updating

### DISCUSSION THREAD

**Discussion by WardPearce** [encode/starlette#1676] [Potential Issue] [answered]
Recently after making some minor changes to my project, I found that my workflow for my unit tests stopped passing but the tests on my local machine still passed. After upgrading Starlette to `0.20.1` on my local machine the unit tests began to fail.

After doing a bit of digging, it appears none of my background tasks are running anymore. When replacing the background tasks with just the function they call in the background, the unit tests past.

This is also happening outside of unit tests too, when spinning up my development environment locally. The API doesn't function as it should.

Also after downgrading back to `0.20.0` everything works as expected.

- [Source code](https://github.com/WardPearce/paaster/blob/Development/paaster-backend/paaster/routes/http/paste.py#L153)
- Python version: `3.10.4`
- Starlette version: `0.20.1`

**Comment by Kludex** [comment, marked as answer]
Do you use BaseHTTPMiddleware?

**Reply by WardPearce** [reply]
Nope.

**Reply by Kludex** [reply]
Are you able to reproduce on a minimal setup that I can run?

**Reply by WardPearce** [reply]
Yeah tried replicating it with a simple one page application but was unable to. Don't believe my project does anything funky. So I'm completely confused why `0.20.1` appears to break things.

Must be something to do with `0.20.1` if that's the only changing variable between it working or not correct? Especially that this is happening across 2 different python versions on different hardware (My workflow uses Py 3.9)

**Reply by Kludex** [reply]
slowapi uses BaseHTTPMiddleware.

**Reply by Kludex** [reply]
You can't use background tasks with BaseHTTPMiddleware. 😟

**Reply by WardPearce** [reply]
> slowapi uses BaseHTTPMiddleware.

Oh that would make sense!

**Reply by WardPearce** [reply]
> You can't use background tasks with BaseHTTPMiddleware. worried

Is this new to `0.20.1`?

**Reply by Kludex** [reply]
It's not... It's an old issue.

I don't know why you find it now, but that's why the background tasks don't run.

**Reply by WardPearce** [reply]
Odd, yea interesting that they're running on `0.20.0`

**Reply by Kludex** [reply]
Can you add an sleep of 1 sec on the background task and see if it's still the case?

**Reply by WardPearce** [reply]
Sure, I'll do that tmrw. Spent today porting my project over to FastAPI.

**Reply by Kludex** [reply]
Cool! If you need something, just in case, we have a discord server for FastAPI: https://discord.com/invite/VQjSZaeJmf :)

**Reply by WardPearce** [reply]
Awesome thanks, FastAPI is great. Shame it doesn't have class based routes. Often the reasons why I stick with starlette for smaller projects.
